### PR TITLE
Strategy pattern

### PR DESCRIPTION
**Pull request by anuragdhingra**
- Add strategy pattern example
- Add strategy.md

### CODE REVIEW

File: strategy/main.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package main
+
+import "fmt"
+
+// QuackStrategy interface
+type QuackStrategy interface {
```
Comment (amanjain97) [inline strategy/main.go:6]: https://golang.org/doc/effective_go.html#interface-names

**Review by amanjain97 — APPROVED** [review]

File: strategy/main.go (hunk 1 above)
Comment (anuragdhingra) [reply strategy/main.go:6]: Want me to name it `QuackStrateger`, `DisplayStrateger` and `FlyStrateger`? Sound a bit odd to me.

Comment (amanjain97) [reply strategy/main.go:6]: It’s just a suggestion. 
Quacker. But that also sounds odd.

Comment (anuragdhingra) [reply strategy/main.go:6]: Named it so to be in-sync with the explanation from the video. Let's maybe keep it that way?

**Comment by anuragdhingra** [comment]
Sorry, re-requested by mistake, merging it so you can rebase your branch directly over `master`.

File: strategy/main.go (hunk 1 above)
Comment (amanjain97) [reply strategy/main.go:6]: Yes yes. Sure.

**Pull request merged**